internal/querynode/query_node.go

```diff
@@ -205,6 +206,14 @@ func (node *QueryNode) Init() error {
 			node.tSafeReplica,
 		)
 
+		node.loader = newSegmentLoader(node.queryNodeLoopCtx,
```

Comment:
the init function should use doSlow rather than initOnce.do, otherwise init becomes async and if any errors happen the query will hang

Reply:
`Once.Do` makes sure all the other calls wait until the first one returns. In the other words, if `Init` process takes 10 seconds and there are 10 invocations of `Init`. Then all of those 10 calls will wait for 10 seconds and there will be only one execution. If any error happens during the `Init` process, making sure query node quit will do the trick.

Reply:
Return initError here